aws_lambda_builders/workflows/custom_make/workflow.py

```diff
@@ -45,17 +44,43 @@ def __init__(self, source_dir, artifacts_dir, scratch_dir, manifest_path, runtim
 
         subprocess_make = SubProcessMake(make_exe=self.binaries["make"].binary_path, osutils=self.os_utils)
 
+        build_in_source = kwargs.get("build_in_source")
+        # Don't build in source by default (backwards compatibility)
+        if build_in_source is None:
+            build_in_source = False
+
+        working_directory = self._get_working_directory(options, source_dir, scratch_dir, build_in_source)
+
         make_action = CustomMakeAction(
             artifacts_dir,
-            scratch_dir,
             manifest_path,
             osutils=self.os_utils,
             subprocess_make=subprocess_make,
             build_logical_id=build_logical_id,
             working_directory=working_directory,
         )
 
-        self.actions = [CopySourceAction(source_dir, scratch_dir, excludes=self.EXCLUDED_FILES), make_action]
+        self.actions = []
+
+        if not self.build_in_source:
+            # if we're building on scratch_dir, we have to first copy the source there
+            self.actions.append(CopySourceAction(source_dir, scratch_dir, excludes=self.EXCLUDED_FILES))
+
+        self.actions.append(make_action)
+
+    def _get_working_directory(self, options, source_dir, scratch_dir, build_in_source):
```

Comment:
nit: Can we add type hints for newer functions. We're trying to be better about it going forward in this repo.

Reply:
for sure, will do.